tests/unit/multi_tenancy.cpp

```diff
@@ -0,0 +1,390 @@
+    TearDown();
+    min_mg.emplace(config);
+  }
+
+  void TearDown() override {
+    min_mg.reset();
+    if (std::filesystem::exists(data_directory)) std::filesystem::remove_all(data_directory);
+  }
+
+  auto NewInterpreter() { return min_mg->NewInterpreter(); }
+
+  auto &DBMS() { return min_mg->dbms; }
+
+  std::optional<MinMemgraph> min_mg;
+};
+
+TEST_F(MultiTenantTest, SimpleCreateDrop) {
+  // 1) Create multiple interpreters with the default db
+  // 2) Create multiple databases using both
+  // 3) Drop databases while the other is using
+
+  // 1
+  auto interpreter1 = this->NewInterpreter();
+  auto interpreter2 = this->NewInterpreter();
+
+  // 2
+  auto create = [&](auto &interpreter, const std::string &name, bool success) {
+    RunMtQuery(interpreter, "CREATE DATABASE " + name,
+               success ? ("Successfully created database " + name) : (name + " already exists."));
+  };
+
+  create(interpreter1, "db1", true);
+  create(interpreter1, "db1", false);
+  create(interpreter2, "db1", false);
+  create(interpreter2, "db2", true);
+  create(interpreter1, "db2", false);
+  create(interpreter2, "db3", true);
+  create(interpreter2, "db4", true);
+
+  // 3
+  auto use = [&](auto &interpreter, const std::string &name, std::string_view res) {
+    RunMtQuery(interpreter, "USE DATABASE " + name, res);
+  };
+
+  auto drop = [&](auto &interpreter, const std::string &name, std::string_view res) {
+    RunMtQuery(interpreter, "DROP DATABASE " + name, res);
+  };
+
+  use(interpreter1, "db2", "Using db2");
+  use(interpreter1, "db2", "Already using db2");
+  use(interpreter2, "db2", "Using db2");
+  use(interpreter1, "db4", "Using db4");
+
+  ASSERT_THROW(drop(interpreter1, memgraph::dbms::kDefaultDB.data(), ""),
+               memgraph::query::QueryRuntimeException);  // default db
+
+  drop(interpreter1, "db1", "Successfully deleted db1");
+  ASSERT_THROW(drop(interpreter2, "db1", ""), memgraph::query::QueryRuntimeException);  // No db1
+  ASSERT_THROW(drop(interpreter1, "db1", ""), memgraph::query::QueryRuntimeException);  // No db1
+
+  ASSERT_THROW(drop(interpreter1, "db2", ""), memgraph::query::QueryRuntimeException);  // i2 using db2
+  ASSERT_THROW(drop(interpreter1, "db4", ""), memgraph::query::QueryRuntimeException);  // i1 using db4
+}
+
+TEST_F(MultiTenantTest, DbmsNewTryDelete) {
+  // 1) Create multiple interpreters with the default db
+  // 2) Create multiple databases using dbms
+  // 3) Try delete databases while the interpreters are using them
+
+  // 1
+  auto interpreter1 = this->NewInterpreter();
+  auto interpreter2 = this->NewInterpreter();
+
+  // 2
+  auto &dbms = DBMS();
+  ASSERT_FALSE(dbms.New("db1").HasError());
+  ASSERT_FALSE(dbms.New("db2").HasError());
+  ASSERT_FALSE(dbms.New("db3").HasError());
+  ASSERT_FALSE(dbms.New("db4").HasError());
+
+  // 3
+  auto use = [&](auto &interpreter, const std::string &name, std::string_view res) {
+    RunMtQuery(interpreter, "USE DATABASE " + name, res);
+  };
```

Comment:
This can be maybe part of MultiTenantTest function? It is used also in test above